In [1]:
import numpy as np
%matplotlib inline
import matplotlib.pyplot as plt
import pandas as pd
from dateutil import parser

Connecting the data in the notebook

In [2]:
df = pd.read_csv("C:/Users/lalac/OneDrive/Documents/Open Avenues project/sample - in.csv")
df.head()

tweet_id     author_id  inbound                      created_at  \
0    119237        105834     True  Wed Oct 11 06:55:44 +0000 2017   
1    119238  ChaseSupport    False  Wed Oct 11 13:25:49 +0000 2017   
2    119239        105835     True  Wed Oct 11 13:00:09 +0000 2017   
3    119240  VirginTrains    False  Tue Oct 10 15:16:08 +0000 2017   
4    119241        105836     True  Tue Oct 10 15:17:21 +0000 2017   

                                                text response_tweet_id  \
0  @AppleSupport causing the reply to be disregar...            119236   
1  @105835 Your business means a lot to us. Pleas...               NaN   
2  @76328 I really hope you all change but I'm su...            119238   
3  @105836 LiveChat is online at the moment - htt...            119241   
4  @VirginTrains see attached error message. I've...            119243   

   in_response_to_tweet_id  
0                      NaN  
1                 119239.0  
2                      NaN  
3                 119242.0  
4                 119240.0

<b>Create a few points on your understanding and explanation of each column:</b>
1. Automatic id
2. id of the person who is contacting and the company who is responding
3. inbound true if it's a user message and false if it's a message from the company
4. date and hour of the message
5. text sent or received
6 and 7. conversation between company and customer based on the tweet_id

<b>Create a few points on your understanding of the data:</b>
* Chatbot for people asking to solve their issues with different companies and the responses that are given by the supoort team

<b>Understanding the term Data Exploration</b>
* "Data exploration is the first step of data analysis used to explore and visualize data to uncover insights from the start or identify areas or patterns to dig into more"

<b>Data Exploration</b>

In [3]:
df.shape #Number of rows and columns

(93, 7)

In [4]:
df.info() #data type for each column 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93 entries, 0 to 92
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   tweet_id                 93 non-null     int64  
 1   author_id                93 non-null     object 
 2   inbound                  93 non-null     bool   
 3   created_at               93 non-null     object 
 4   text                     93 non-null     object 
 5   response_tweet_id        65 non-null     object 
 6   in_response_to_tweet_id  68 non-null     float64
dtypes: bool(1), float64(1), int64(1), object(4)
memory usage: 4.6+ KB


In [5]:
df.describe()

tweet_id  in_response_to_tweet_id
count      93.000000                68.000000
mean   119285.451613            119285.676471
std        28.314045                28.910795
min    119237.000000            119239.000000
25%    119262.000000            119259.750000
50%    119285.000000            119284.500000
75%    119309.000000            119311.500000
max    119335.000000            119334.000000

In [6]:
df.isnull().sum() #I'm unsure if we need to replace the null values

tweet_id                    0
author_id                   0
inbound                     0
created_at                  0
text                        0
response_tweet_id          28
in_response_to_tweet_id    25
dtype: int64

In [7]:
#In case we need it, here we are replacing Null values with the word Null
#df['response_tweet_id'].replace(np.nan,'Null',inplace=True)
#df['in_response_to_tweet_id'].replace(np.nan,'Null',inplace=True)

In [8]:
#df.isnull().sum()

In [9]:
#Duplicate values
duplicate= df.duplicated()
print(duplicate.sum())
df[duplicate] #there aren't rows with same values

0


Empty DataFrame
Columns: [tweet_id, author_id, inbound, created_at, text, response_tweet_id, in_response_to_tweet_id]
Index: []

Inserting the new dataset

In [10]:
df2 = pd.read_csv("C:/Users/lalac/OneDrive/Documents/Open Avenues project/twcs.csv")
df2.head()

tweet_id   author_id  inbound                      created_at  \
0         1  sprintcare    False  Tue Oct 31 22:10:47 +0000 2017   
1         2      115712     True  Tue Oct 31 22:11:45 +0000 2017   
2         3      115712     True  Tue Oct 31 22:08:27 +0000 2017   
3         4  sprintcare    False  Tue Oct 31 21:54:49 +0000 2017   
4         5      115712     True  Tue Oct 31 21:49:35 +0000 2017   

                                                text response_tweet_id  \
0  @115712 I understand. I would like to assist y...                 2   
1      @sprintcare and how do you propose we do that               NaN   
2  @sprintcare I have sent several private messag...                 1   
3  @115712 Please send us a Private Message so th...                 3   
4                                 @sprintcare I did.                 4   

   in_response_to_tweet_id  
0                      3.0  
1                      1.0  
2                      4.0  
3                      5.0  
4                      6.0

In [11]:
df2.shape #Number of rows and columns

(2811774, 7)

In [12]:
df2.info() #datatype

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2811774 entries, 0 to 2811773
Data columns (total 7 columns):
 #   Column                   Dtype  
---  ------                   -----  
 0   tweet_id                 int64  
 1   author_id                object 
 2   inbound                  bool   
 3   created_at               object 
 4   text                     object 
 5   response_tweet_id        object 
 6   in_response_to_tweet_id  float64
dtypes: bool(1), float64(1), int64(1), object(4)
memory usage: 131.4+ MB


In [13]:
df2.isnull().sum() #finding null values

tweet_id                         0
author_id                        0
inbound                          0
created_at                       0
text                             0
response_tweet_id          1040629
in_response_to_tweet_id     794335
dtype: int64

In [14]:
duplicate2= df2.duplicated()
print(duplicate2.sum())
df2[duplicate2]

0


Empty DataFrame
Columns: [tweet_id, author_id, inbound, created_at, text, response_tweet_id, in_response_to_tweet_id]
Index: []

3 ways that should be cleaned:
1. Deleting information that isn't relevant( for example at created_at we only need the date and time)
2. Finding the Null values and understand what are they telling us and replace with a value
3. Find duplicates or group the data by tweet id


<b>How many are inbound and outbound responses?</b>

In [15]:
inbound = df[df['inbound'] ==True]
inbound.count()

tweet_id                   49
author_id                  49
inbound                    49
created_at                 49
text                       49
response_tweet_id          44
in_response_to_tweet_id    24
dtype: int64

In [16]:
outbound = 93 - inbound.count()
outbound 

tweet_id                   44
author_id                  44
inbound                    44
created_at                 44
text                       44
response_tweet_id          49
in_response_to_tweet_id    69
dtype: int64

In [17]:
inbound2 = df2[df2['inbound']==True]
inbound2.count()

tweet_id                   1537843
author_id                  1537843
inbound                    1537843
created_at                 1537843
text                       1537843
response_tweet_id          1303829
in_response_to_tweet_id     750497
dtype: int64

In [18]:
outbound2 = 2811774 - inbound2.count()
outbound2 

tweet_id                   1273931
author_id                  1273931
inbound                    1273931
created_at                 1273931
text                       1273931
response_tweet_id          1507945
in_response_to_tweet_id    2061277
dtype: int64

For the first dataset 49 are inbound responses and 44 are outbound responses. 
For the second data set 1537873 are inbound and 1273931 are outbound responses

<b>How many are unique tweets?</b>

In [19]:
select_tweet= df['text']
duplicate_tweet=select_tweet.duplicated().sum()
duplicate_tweet

0

In [20]:
select_tweet2= df2['text']
duplicate_tweet2=select_tweet2.duplicated().sum()
duplicate_tweet2

29156

In [21]:
unique_tweet2 = 2811774 - duplicate_tweet2
unique_tweet2

2782618

All the tweets are different base on the text column for first data set but for second there are 2782618 unique tweets

<b>Top 20 tweer users </b>

In [22]:
author_id = df.groupby(['author_id']).count().sort_values(by='tweet_id', ascending = False)
top_20= author_id.head(20)
top_20

tweet_id  inbound  created_at  text  response_tweet_id  \
author_id                                                                 
AppleSupport           13       13          13    13                  3   
Tesco                   8        8           8     8                  6   
SpotifyCares            8        8           8     8                  6   
105855                  4        4           4     4                  3   
VirginTrains            4        4           4     4                  3   
105847                  4        4           4     4                  4   
105840                  4        4           4     4                  4   
105848                  3        3           3     3                  1   
105836                  3        3           3     3                  3   
British_Airways         3        3           3     3                  1   
105856                  3        3           3     3                  3   
105854                  2        2           2     2                  1   
105842                  2        2           2     2                  2   
105844                  2        2           2     2                  2   
105857                  2        2           2     2                  2   
105850                  2        2           2     2                  1   
82476                   1        1           1     1                  1   
Ask_Spectrum            1        1           1     1                  1   
HPSupport               1        1           1     1                  0   
ChaseSupport            1        1           1     1                  0   

                 in_response_to_tweet_id  
author_id                                 
AppleSupport                          13  
Tesco                                  8  
SpotifyCares                           8  
105855                                 3  
VirginTrains                           4  
105847                                 3  
105840                                 3  
105848                                 2  
105836                                 3  
British_Airways                        3  
105856                                 2  
105854                                 1  
105842                                 1  
105844                                 1  
105857                                 1  
105850                                 1  
82476                                  0  
Ask_Spectrum                           1  
HPSupport                              1  
ChaseSupport                           1

In [23]:
author_id2= df2.groupby(['author_id']).count().sort_values(by='tweet_id', ascending = False)
top_202= author_id2.head(20)
top_202

tweet_id  inbound  created_at    text  response_tweet_id  \
author_id                                                                   
AmazonHelp         169840   169840      169840  169840              85274   
AppleSupport       106860   106860      106860  106860              31564   
Uber_Support        56270    56270       56270   56270              18036   
SpotifyCares        43265    43265       43265   43265              13786   
Delta               42253    42253       42253   42253              12014   
Tesco               38573    38573       38573   38573              11148   
AmericanAir         36764    36764       36764   36764              14556   
TMobileHelp         34317    34317       34317   34317               9759   
comcastcares        33031    33031       33031   33031               7625   
British_Airways     29361    29361       29361   29361              10099   
SouthwestAir        28977    28977       28977   28977               8580   
VirginTrains        27817    27817       27817   27817              15235   
Ask_Spectrum        25860    25860       25860   25860               7693   
XboxSupport         24557    24557       24557   24557              10241   
sprintcare          22381    22381       22381   22381               7334   
hulu_support        21872    21872       21872   21872               7705   
sainsburys          19466    19466       19466   19466               8172   
GWRHelp             19364    19364       19364   19364              10915   
AskPlayStation      19098    19098       19098   19098               8065   
ChipotleTweets      18749    18749       18749   18749               6010   

                 in_response_to_tweet_id  
author_id                                 
AmazonHelp                        169287  
AppleSupport                      106719  
Uber_Support                       56261  
SpotifyCares                       43243  
Delta                              42197  
Tesco                              38501  
AmericanAir                        36598  
TMobileHelp                        34287  
comcastcares                       33007  
British_Airways                    29315  
SouthwestAir                       28889  
VirginTrains                       27522  
Ask_Spectrum                       25807  
XboxSupport                        24341  
sprintcare                         22335  
hulu_support                       21783  
sainsburys                         19417  
GWRHelp                            19294  
AskPlayStation                     18694  
ChipotleTweets                     18612

# Data Cleaning

<b>1. Cleaning tweets that are not in_response to customers tweets at first</b>

In [24]:
#Tweets by company where inbound = false
df_comp= df2[df2['inbound'] == False]
df_comp

tweet_id       author_id  inbound                      created_at  \
0               1      sprintcare    False  Tue Oct 31 22:10:47 +0000 2017   
3               4      sprintcare    False  Tue Oct 31 21:54:49 +0000 2017   
5               6      sprintcare    False  Tue Oct 31 21:46:24 +0000 2017   
7              11      sprintcare    False  Tue Oct 31 22:10:35 +0000 2017   
9              15      sprintcare    False  Tue Oct 31 20:03:31 +0000 2017   
...           ...             ...      ...                             ...   
2811764   2987943  AirAsiaSupport    False  Wed Nov 22 07:54:57 +0000 2017   
2811766    139628    ArgosHelpers    False  Wed Nov 22 08:03:26 +0000 2017   
2811767   2987945    VirginTrains    False  Wed Nov 22 08:27:34 +0000 2017   
2811769   2987947      sprintcare    False  Wed Nov 22 08:43:51 +0000 2017   
2811772   2987949          AldiUK    False  Wed Nov 22 08:31:24 +0000 2017   

                                                      text response_tweet_id  \
0        @115712 I understand. I would like to assist y...                 2   
3        @115712 Please send us a Private Message so th...                 3   
5        @115712 Can you please send us a private messa...               5,7   
7        @115713 This is saddening to hear. Please shoo...               NaN   
9        @115713 We understand your concerns and we'd l...                12   
...                                                    ...               ...   
2811764  @823868 Sorry but kindly try to clear browser,...               NaN   
2811766  @134664 Can you Dm us your order number and we...               NaN   
2811767  @524544 That's a Peak service. The 09:56 is th...               NaN   
2811769  @823869 Hey, we'd be happy to look into this f...               NaN   
2811772  @823870 Sounds delicious, Sarah! 😋 https://t.c...               NaN   

         in_response_to_tweet_id  
0                            3.0  
3                            5.0  
5                            8.0  
7                           12.0  
9                           16.0  
...                          ...  
2811764                2987944.0  
2811766                 139627.0  
2811767                2987946.0  
2811769                2987948.0  
2811772                2987950.0  

[1273931 rows x 7 columns]

In [25]:
#Clean rows if there is not a value in in_response_to_tweet_id
df_inbound = df2[(df2["inbound"] == False)]
filtered_df = df_inbound[df_inbound['in_response_to_tweet_id'].notnull()]
filtered_df


tweet_id       author_id  inbound                      created_at  \
0               1      sprintcare    False  Tue Oct 31 22:10:47 +0000 2017   
3               4      sprintcare    False  Tue Oct 31 21:54:49 +0000 2017   
5               6      sprintcare    False  Tue Oct 31 21:46:24 +0000 2017   
7              11      sprintcare    False  Tue Oct 31 22:10:35 +0000 2017   
9              15      sprintcare    False  Tue Oct 31 20:03:31 +0000 2017   
...           ...             ...      ...                             ...   
2811764   2987943  AirAsiaSupport    False  Wed Nov 22 07:54:57 +0000 2017   
2811766    139628    ArgosHelpers    False  Wed Nov 22 08:03:26 +0000 2017   
2811767   2987945    VirginTrains    False  Wed Nov 22 08:27:34 +0000 2017   
2811769   2987947      sprintcare    False  Wed Nov 22 08:43:51 +0000 2017   
2811772   2987949          AldiUK    False  Wed Nov 22 08:31:24 +0000 2017   

                                                      text response_tweet_id  \
0        @115712 I understand. I would like to assist y...                 2   
3        @115712 Please send us a Private Message so th...                 3   
5        @115712 Can you please send us a private messa...               5,7   
7        @115713 This is saddening to hear. Please shoo...               NaN   
9        @115713 We understand your concerns and we'd l...                12   
...                                                    ...               ...   
2811764  @823868 Sorry but kindly try to clear browser,...               NaN   
2811766  @134664 Can you Dm us your order number and we...               NaN   
2811767  @524544 That's a Peak service. The 09:56 is th...               NaN   
2811769  @823869 Hey, we'd be happy to look into this f...               NaN   
2811772  @823870 Sounds delicious, Sarah! 😋 https://t.c...               NaN   

         in_response_to_tweet_id  
0                            3.0  
3                            5.0  
5                            8.0  
7                           12.0  
9                           16.0  
...                          ...  
2811764                2987944.0  
2811766                 139627.0  
2811767                2987946.0  
2811769                2987948.0  
2811772                2987950.0  

[1266942 rows x 7 columns]

Only 1266942 of the tweets were responses to customers tweets from companies

<b> 2. Finding outliners </b>

In [26]:
#for questions
print(type(filtered_df['author_id']))

<class 'pandas.core.series.Series'>


In [27]:
#for questions isn't supposed to be 
filtered= filtered_df[(filtered_df["author_id"] == "115911")]
filtered

Empty DataFrame
Columns: [tweet_id, author_id, inbound, created_at, text, response_tweet_id, in_response_to_tweet_id]
Index: []

In [28]:
#getting all the customer inbound = true, and finding the ones with a most tweets
customers= df2[(df2["inbound"] == True)]
false_customer = customers.groupby(['author_id']).count().sort_values(by='tweet_id', ascending = False)
num_tweets_false_customer= false_customer['in_response_to_tweet_id']
num_tweets_false_customer


author_id
115911    1124
120576    1002
115913     179
116230     432
169172     446
          ... 
549508       0
549507       0
345292       0
549505       0
263844       0
Name: in_response_to_tweet_id, Length: 702669, dtype: int64

In [29]:
num_tweets_false_customer.mean()

1.0680661876359994

In [30]:
num_tweets_false_customer.median()

0.0

In [31]:
# outliner max num the tweets from false customers
num_tweets_false_customer.max()


1124

In [32]:
#number of author_id that coudl be companies
false_customer.head(20)

tweet_id  inbound  created_at  text  response_tweet_id  \
author_id                                                           
115911         1286     1286        1286  1286               1286   
120576         1010     1010        1010  1010                669   
115913          563      563         563   563                561   
116230          454      454         454   454                454   
169172          448      448         448   448                447   
117627          406      406         406   406                302   
115888          332      332         332   332                332   
116136          295      295         295   295                290   
116421          276      276         276   276                218   
115722          252      252         252   252                252   
115714          250      250         250   250                250   
115850          243      243         243   243                243   
115990          243      243         243   243                243   
121239          210      210         210   210                130   
203476          197      197         197   197                128   
115725          194      194         194   194                194   
115798          181      181         181   181                179   
127296          176      176         176   176                 98   
170351          174      174         174   174                 35   
169916          172      172         172   172                172   

           in_response_to_tweet_id  
author_id                           
115911                        1124  
120576                        1002  
115913                         179  
116230                         432  
169172                         446  
117627                         382  
115888                           1  
116136                         247  
116421                         172  
115722                         228  
115714                         114  
115850                          91  
115990                         232  
121239                         152  
203476                         175  
115725                         147  
115798                         178  
127296                         153  
170351                         174  
169916                         172

In [33]:
# for questions how do you know if they are customers or companies?
# exploering some of the false customers
cust_115911 = df2[df2['author_id'] == '115911']
cust_115911.head(5)
#this could be a company

tweet_id author_id  inbound                      created_at  \
622       1068    115911     True  Tue Oct 31 22:19:12 +0000 2017   
624       1070    115911     True  Tue Oct 31 22:16:12 +0000 2017   
626       1072    115911     True  Thu Oct 26 21:13:40 +0000 2017   
1346      1950    115911     True  Tue Oct 31 22:27:18 +0000 2017   
1348      1952    115911     True  Tue Oct 31 13:00:17 +0000 2017   

                                                   text  \
622   @115910 No sir! You can still preorder at http...   
624                     @115910 Go for it Trevor *ErikS   
626   Be ready for your next iPhone. Join our iPhone...   
1346  @116149 I mean...it does say to "submit your c...   
1348  Happy Halloween! Submit your costume pics with...   

                                      response_tweet_id  \
622                                                1067   
624                                                1069   
626   1071,1073,1074,1075,1076,1077,1078,1079,1080,1...   
1346                                               1949   
1348  1953,1954,1955,1956,1957,1958,1959,1960,1961,1...   

      in_response_to_tweet_id  
622                    1069.0  
624                    1071.0  
626                       NaN  
1346                   1951.0  
1348                      NaN

In [34]:
cust_176402 = df2[df2['author_id'] == '176402']
cust_176402.head(5)
#this could be a customer

tweet_id author_id  inbound                      created_at  \
218496    254108    176402     True  Thu Oct 05 22:32:41 +0000 2017   
220081    255805    176402     True  Thu Oct 05 21:29:36 +0000 2017   
224830    260869    176402     True  Thu Oct 05 21:33:22 +0000 2017   
224833    260873    176402     True  Thu Oct 05 21:34:06 +0000 2017   
224834    260875    176402     True  Thu Oct 05 21:34:23 +0000 2017   

                                                     text response_tweet_id  \
218496  @AppleSupport @176401 I have the same problem ...               NaN   
220081  @AppleSupport @176842 11.0.2 stinks Apple just...               NaN   
224830  @AppleSupport @178108 Kevin I honestly think t...               NaN   
224833  @AppleSupport @178109 Over two weeks people ar...            260875   
224834  @AppleSupport @178109 They want us all to go b...               NaN   

        in_response_to_tweet_id  
218496                 254107.0  
220081                 255804.0  
224830                 260868.0  
224833                 260872.0  
224834                 260873.0

In [35]:
#top 20 by number of tweets

<b> Top tweeter users (customers) by number of tweets </b>

In [36]:
#false_customer.head(20) - the ones that are companies. For question

<b> Top authors by response </b>

In [37]:
#is it by response or just by number of tweets> for question
top_authors_by_response = filtered_df.groupby(['author_id']).count().sort_values(by='response_tweet_id', ascending = False)
top_authors_by_response

tweet_id  inbound  created_at    text  response_tweet_id  \
author_id                                                                  
AmazonHelp        169287   169287      169287  169287              84721   
AppleSupport      106719   106719      106719  106719              31423   
Uber_Support       56261    56261       56261   56261              18027   
VirginTrains       27522    27522       27522   27522              14940   
AmericanAir        36598    36598       36598   36598              14390   
...                  ...      ...         ...     ...                ...   
TwitterSupport      1278     1278        1278    1278                 86   
CarlsJr              183      183         183     183                 42   
JackBox              255      255         255     255                 38   
HotelTonightCX       152      152         152     152                 30   
AskDSC               210      210         210     210                 29   

                in_response_to_tweet_id  
author_id                                
AmazonHelp                       169287  
AppleSupport                     106719  
Uber_Support                      56261  
VirginTrains                      27522  
AmericanAir                       36598  
...                                 ...  
TwitterSupport                     1278  
CarlsJr                             183  
JackBox                             255  
HotelTonightCX                      152  
AskDSC                              210  

[108 rows x 6 columns]

<b> Tweets by date </b>

In [38]:
##changing the created_at format to a datetime format
df2['created_at'] = pd.to_datetime(df['created_at'])
df2

tweet_id   author_id  inbound                created_at  \
0               1  sprintcare    False 2017-10-11 06:55:44+00:00   
1               2      115712     True 2017-10-11 13:25:49+00:00   
2               3      115712     True 2017-10-11 13:00:09+00:00   
3               4  sprintcare    False 2017-10-10 15:16:08+00:00   
4               5      115712     True 2017-10-10 15:17:21+00:00   
...           ...         ...      ...                       ...   
2811769   2987947  sprintcare    False                       NaT   
2811770   2987948      823869     True                       NaT   
2811771   2812240      121673     True                       NaT   
2811772   2987949      AldiUK    False                       NaT   
2811773   2987950      823870     True                       NaT   

                                                      text response_tweet_id  \
0        @115712 I understand. I would like to assist y...                 2   
1            @sprintcare and how do you propose we do that               NaN   
2        @sprintcare I have sent several private messag...                 1   
3        @115712 Please send us a Private Message so th...                 3   
4                                       @sprintcare I did.                 4   
...                                                    ...               ...   
2811769  @823869 Hey, we'd be happy to look into this f...               NaN   
2811770  @115714 wtf!? I’ve been having really shitty s...           2987947   
2811771  @143549 @sprintcare You have to go to https://...               NaN   
2811772  @823870 Sounds delicious, Sarah! 😋 https://t.c...               NaN   
2811773  @AldiUK  warm sloe gin mince pies with ice cre...   2987951,2987949   

         in_response_to_tweet_id  
0                            3.0  
1                            1.0  
2                            4.0  
3                            5.0  
4                            6.0  
...                          ...  
2811769                2987948.0  
2811770                      NaN  
2811771                2812239.0  
2811772                2987950.0  
2811773                      NaN  

[2811774 rows x 7 columns]

In [39]:
tweets_by_date = df2[['created_at', 'tweet_id']]
tweets_by_date.isnull().sum()

created_at    2811681
tweet_id            0
dtype: int64

In [40]:
tweets_by_dat= tweets_by_date.groupby(['created_at']).count()
tweets_by_dat

tweet_id
created_at                         
2017-10-10 10:13:19+00:00         1
2017-10-10 15:09:00+00:00         1
2017-10-10 15:16:08+00:00         1
2017-10-10 15:17:21+00:00         1
2017-10-10 15:25:14+00:00         1
...                             ...
2017-10-11 16:28:34+00:00         1
2017-10-12 10:25:35+00:00         1
2017-10-12 11:43:57+00:00         1
2017-10-12 12:04:21+00:00         1
2017-10-12 12:09:13+00:00         1

[93 rows x 1 columns]